brain/s3/bbon/behavior_narrator.py

```
@@ -0,0 +1,273 @@
+            # Clamp coordinates within bounds
+            width = max(0, min(img.width - 1, width))
+            height = max(0, min(img.height - 1, height))
+
+            def place_text(label, color):
+                bbox = draw.textbbox((0, 0), label, font=font)
+                text_w, text_h = (
+                    bbox[2] - bbox[0],
+                    bbox[3] - bbox[1],
+                )  # Measure text size
+                offset_x, offset_y = -5, 5  # Default offset
+                if width + offset_x + text_w > img.width:  # Out of bounds on right
```

Ruff (0.14.6)

Function definition does not bind loop variable `width`

(B023)